src/index.js

```diff
@@ -169,7 +169,21 @@ class MagicalScroll {
     this.scrollTop = 0;
     this.elements = [];
 
-    window.requestAnimationFrame(() => this.refresh(true));
+    // resize event
+    let timeout = null;
+    const resizeObserver = new ResizeObserver(() => {
+      if (timeout) {
+        clearTimeout(timeout);
+      }
+      timeout = setTimeout(() => {
+        this.elements.forEach(this.preProcess.bind(this));
+        this.refresh();
+      }, 16);
```

Comment:
I would suggest to keep `requestAnimationFrame`. Define a flag for resizing `let needResize = false` and turn it to true upon page resize event. Finally call the refresh in the `requestAnimationFrame` if `needResize === true`.
```javascript
let needResize = false;
window.requestAnimationFrame(() => {
  if (needResize) {
    this.elements.forEach(this.preProcess.bind(this));
    this.refresh(true);
  }
});
const resizeObserver = new ResizeObserver(() => {
  needResize = true;
});
```

Reply:
but `requestAnimationFrame` will be called about 30 times per second, although window doesn't scroll, so I think `requestAnimationFrame` is too overhead.
I using setTimeout in ResizeObserver's callback to prevent callback be called too many time when resizing. If I set animation time to 1s, the callback will be called 1 * 30 times, but I only need to refresh 1 time when resized, so I decided to add setTimeout in the callback.

Reply:
It won't be called since `needRefresh` is false, so your concern is actually not going to happen I guess. But I'm okay with that solution.